lib/py/src/transport/sslcompat.py

```diff
@@ -66,6 +66,14 @@ def legacy_validate_callback(cert, hostname):
 
```

Comment:
As far as I can see, this does not change any of our docker image behaviors for python:
SSLv23 with NO_SSL* is being used for python 3 and debian python 2.

About defaulting to TLSv1.2, Python < 2.7.9 and C# with .NET 3.5 compat do not seem to have TLS 1.1 and 1.2 at all.
The highest versions listed in doc is TLSv1.0.

For C#, once dropping .NET 3.5, it would be able to support TLS 1.0, 1.1 and 1.2 at once because SslProtocols is an explicit "flags" enum that can be combined with pipe like `Tls12 | Tls11 | Tls`.
Though I agree that it's not possible to do forward compatible "SSLv23 minus SSL 2 and 3".